crates/pallet-domains/src/tests.rs

```diff
@@ -1124,3 +1189,287 @@ fn test_type_with_default_nonce_encode() {
+            },
+            // this version at this block must exist due to the change in version
+            // in the next upgrade
+            true,
+        ),
+        // version change
+        // version from 131..150
+        (
+            150u32,
+            MockBundleAndExecutionReceiptVersion {
+                bundle_version: MockBundleVersion::V1,
+                execution_receipt_version: MockExecutionReceiptVersion::V1,
+            },
+            // this version at this block will not exist since next upgrade
+            // carries same version
+            false,
+        ),
+        // same version as previous
+        // version from 151..155
+        (
+            155u32,
+            MockBundleAndExecutionReceiptVersion {
+                bundle_version: MockBundleVersion::V1,
+                execution_receipt_version: MockExecutionReceiptVersion::V1,
+            },
+            // this version at this block must exist due to the change in version
+            // in the next upgrade
+            true,
+        ),
+        // version change
+        // version from 156..160
+        (
+            160u32,
+            MockBundleAndExecutionReceiptVersion {
+                bundle_version: MockBundleVersion::V2,
+                execution_receipt_version: MockExecutionReceiptVersion::V1,
+            },
+            // this version at this block must exist due to the change in version
+            // in the next upgrade
+            true,
+        ),
+        // version change
+        // version from 161..200
+        (
+            200u32,
+            MockBundleAndExecutionReceiptVersion {
+                bundle_version: MockBundleVersion::V2,
+                execution_receipt_version: MockExecutionReceiptVersion::V2,
+            },
+            // this version at this block will not exist since next upgrade
+            // carries same version
+            false,
+        ),
+        // same version
+        // version from 201..250
+        (
+            250u32,
+            MockBundleAndExecutionReceiptVersion {
+                bundle_version: MockBundleVersion::V2,
+                execution_receipt_version: MockExecutionReceiptVersion::V2,
+            },
+            // this version at this block will not exist since next upgrade
+            // carries same version
+            false,
+        ),
+        // same version
+        // version from 251..300
+        (
+            300u32,
+            MockBundleAndExecutionReceiptVersion {
+                bundle_version: MockBundleVersion::V2,
+                execution_receipt_version: MockExecutionReceiptVersion::V2,
+            },
+            // this version at this block must exist due to the change in version
+            // in the next upgrade
+            true,
+        ),
+    ]
+}
+
+fn get_mock_version_queries() -> Vec<(u32, MockBundleAndExecutionReceiptVersion)> {
+    vec![
```

Comment:
It is also helpful if tests and test helper methods have short comments, so we know what you intend them to do.